<?xml version="1.0"?>
<page id="Gio.FileEnumerator.close_finish"
      type="topic"
      style="method"
      xmlns="http://projectmallard.org/1.0/"
      xmlns:api="http://projectmallard.org/experimental/api/"
      xmlns:ui="http://projectmallard.org/1.0/ui/">
  <info>
    <link xref="Gio.FileEnumerator" group="method" type="guide"/>
    <api:function>
      <api:returns>
        <api:type>Boolean</api:type>
      </api:returns>
      <api:name>g_file_enumerator_close_finish</api:name>
      <api:arg>
        <api:type>Gio.AsyncResult</api:type>
        <api:name>result</api:name>
      </api:arg>
    </api:function>
  </info>
  <title>Gio.FileEnumerator.prototype.close_finish</title>
  <synopsis><code mime="text/x-gjs">
function close_finish(result: Gio.AsyncResult): Boolean {
    // Gjs wrapper for g_file_enumerator_close_finish()
}
  </code></synopsis>
  <p>Finishes closing a file enumerator, started from <link xref="Gio.FileEnumerator.close_async"/>.</p>  <p>If the file enumerator was already closed when <link xref="Gio.FileEnumerator.close_async"/>
was called, then this function will report <link xref="Gio.IOErrorEnum"/>.closed in @error, and
return <code>false</code>. If the file enumerator had pending operation when the close
operation was started, then this function will report <link xref="Gio.IOErrorEnum"/>.pending, and
return <code>false</code>.  If @cancellable was not <code>null</code>, then the operation may have been
cancelled by triggering the cancellable object from another thread. If the operation
was cancelled, the error <link xref="Gio.IOErrorEnum"/>.cancelled will be set, and <code>false</code> will be
returned.</p>

<terms>
<item>
<title><code>result</code></title>
  <p>a <link xref="Gio.AsyncResult"/>.</p>
</item>
<item>
<title><code>ok</code></title>

</item>
</terms>

</page>
